cypress/e2e/API/EngageSphere.cy.js

```diff
@@ -0,0 +1,114 @@
+          failOnStatusCode: false 
+        }).then(({ status }) => {
+          expect(status).to.eq(400);
+        });
+      });
+      it('Retorna um erro e código de status 400 ao passar um valor nulo para o parâmetro size', () => {
+        cy.request({
+          method: 'GET',
+          url: `${CUSTOMERS_API_URL}?page=1&limit=10&size=null`,
+          failOnStatusCode: false
+        }).then(({ status }) => {
+          expect(status).to.eq(400);
+        });
+      })
+      it('Retorna um erro e código de status 400 ao passar um boleano', () => {
+        cy.request({
+          method: 'GET',
+          url: `${CUSTOMERS_API_URL}?page=1&limit=true&size=Large%20Enterprise`,
+          failOnStatusCode: false
+        }).then(({ status }) => {
+          expect(status).to.eq(400);
+          //Obs -> não retorna mensagem
+        });
+      });
+    });
+    context('Validação de paginação', () => {
+      const validatePagination = (page) => {
+        cy.request({
+          method: 'GET',
+          url: `${CUSTOMERS_API_URL}?page=${page}&limit=10&size=All`
+        }).then(({ status, body }) => {
+          expect(status).to.eq(200);
+          expect(body.pageInfo).to.be.an('object');
+          expect(body.pageInfo.currentPage).to.eq(`${page}`);
+          if (body.pageInfo.currentPage > body.pageInfo.totalPages) {
+            expect(body.customers).to.be.an('array').that.is.empty;
+          } else {
+            expect(body.customers).to.be.an('array').that.is.not.empty;
+          }
+        });
+      };
+        it('Validação de paginação da primeira página deve retornar 200', () => {
+          validatePagination(1);
+        });
+      
+        it('Validação de paginação da segunda página deve retornar 200', () => {
+          validatePagination(2);
+        }); 
+    
+        it('Validação de paginação da última página deve retornar 200', () => {
+          validatePagination(5);
+        });
+
+        it('Validação de paginação de página vazia deve retornar 200 e nenhum cliente', () => {
+          validatePagination(6);
+        });
+    });
+    context('Recuperação de clientes', () => {
+        it('Recupera clientes com sucesso deve retornar 200', () => {
+            cy.request({
+              method: 'GET',
+              url: `${CUSTOMERS_API_URL}?page=1&limit=10&size=All`,
+            }).then(({ status }) => {
+              expect(status).to.eq(200);
+            });
+          });
+    });
+    context('Recuperação clientes filtrado por size e limit com sucesso', () => {
+      it('Retorna 200 na recuperação do cliente por Small e limite de 10', () => {
+        cy.request({
+          method: 'GET',
+          url: `${CUSTOMERS_API_URL}?page=1&limit=10&size=Small`,
+        }).then(({ status, body }) => {
+          expect(status).to.eq(200);
+          expect(body.customers).to.be.an('array').that.is.not.empty;
+          expect(body.customers.length).to.eq(3);
```

Comment:
Esta _assertion_ é muito específica para o banco de dados em questão, o qual possui 50 customers. E se houvessem mais (ou menos)?
Neste casos, melhor deixar isso claro, por meio de um comentário. Algo assim:

```suggestion
          expect(body.customers.length).to.eq(3); // supondo que há 50 customers no banco de dados.
```